util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/rule/TestElasticsearchClient.java

```diff
@@ -98,6 +98,11 @@ public IndexClient(URLEncodedString primaryIndexName, URLEncodedString writeAlia
 			this.primaryIndexName = primaryIndexName;
 			this.writeAlias = writeAlias;
 			this.readAlias = readAlias;
+
+			// Always delete indexes manipulated through test client after the test;
+			// leaving indexes on the cluster could cause problems
+			// with Amazon OpenSearch Serverless in particular.
+			TestElasticsearchClient.this.registerIndexForCleanup( primaryIndexName );
```

Comment:
I don't think that's worth it:

* Last time I checked dropping an index was relatively fast (a few milliseconds).
* This will only impact a few tests that actually rely on `TestElasticsearchClient`.
* This is a more of a safety net, as Hibernate Search should generally delete these indexes when it stops anyway.